addons/web/static/src/views/form/form_controller.js

```diff
@@ -529,31 +515,30 @@ export class FormController extends Component {
         });
         // FIXME: disable/enable not done in onPagerUpdate
         if (canProceed) {
-            const btns = this.disableButtons();
-            await this.model.load({ resId: false });
-            this.enableButtons(btns);
+            await executeCallbackButton(this.ui.activeElement, () =>
+                this.model.load({ resId: false })
+            );
         }
     }
 
-    async saveButtonClicked(params = {}) {
-        const btns = this.disableButtons();
+    async save(params) {
         const record = this.model.root;
         let saved = false;
-        try {
-            if (this.props.saveRecord) {
-                saved = await this.props.saveRecord(record, params);
-            } else {
-                saved = await record.save(params);
-            }
-        } finally {
-            this.enableButtons(btns);
+        if (this.props.saveRecord) {
+            saved = await this.props.saveRecord(record, params);
+        } else {
+            saved = await record.save(params);
         }
         if (saved && this.props.onSave) {
             this.props.onSave(record, params);
         }
         return saved;
     }
 
+    async saveButtonClicked(params = {}) {
+        await executeCallbackButton(this.ui.activeElement, async () => this.save(params));
```

Comment:
async not necessary (and could return instead of await, so both async not necessary)